frontend/app/routes/$lang+/_protected+/letters+/$id.download.tsx

```diff
@@ -21,7 +23,21 @@ export async function loader({ context: { session }, params, request }: LoaderFu
   const raoidcService = await getRaoidcService();
   await raoidcService.handleSessionValidation(request, session);
 
-  // TODO prevent users from entering any ID in the URL and seeing other users' letters
+  //prevent users from entering any ID in the URL and seeing other users' letters
+  const letters = session.get('letters');
+  const viewLetter = letters.find((letter: { id: string | undefined }) => letter.id === params.id);
```

Comment:
Rather than putting the type here, it should be done when pulling the data from the session.

Reply:
Also, do you need to handle the case where there are no letters found in session?

Reply:
will do it in my next PR